test/test_utils/test_sklearn_wrapper.py

```diff
@@ -0,0 +1,113 @@
+
+from smac.configspace import ConfigurationSpace
+from smac.scenario.scenario import Scenario
+from smac.facade.smac_facade import SMAC
+from smac.utils.sklearn_wrapper import ModelBasedOptimization
+
+
+class SklearnWrapperTest(unittest.TestCase):
+
+    @staticmethod
+    def _config_space_to_param_grid(config_space):
+        ''' Converts config space into the scikit learn param grid format. Could be useful utility function, but
+        does not contain the right functionality yet.
+        '''
+        param_grid = {}
+
+        for hyperparameter in config_space.get_hyperparameters():
+            if isinstance(hyperparameter, CategoricalHyperparameter):
+                param_grid[hyperparameter.name] = hyperparameter.choices
+            elif isinstance(hyperparameter, UniformIntegerHyperparameter):
+                if hyperparameter.log:
+                    raise ValueError('Parameter logscale unimplemented: %s' + str(hyperparameter.__class__.__name__))
+                param_grid[hyperparameter.name] = list(range(hyperparameter.lower, hyperparameter.upper+1))
+            else:
+                raise ValueError('Parameter type unimplemented: %s' + str(hyperparameter.__class__.__name__))
+
+        return param_grid
+
+    def _compare_with_smac(self, X, y, classifier, config_space, random_seed, n_iter):
+        """
+        Checks whether a vanilla SMAC run and the wrapped SMAC run yield the same:
+         - incumbent
+         - evaluation on the incumbent
+        """
+        # important for testing: cv is None and y is classification
+        mbo_wrapper = ModelBasedOptimization(classifier, self._config_space_to_param_grid(config_space),
+                                             random_state=random_seed, verbose=3, n_iter=n_iter)
+        # make a partial for running SMAC under same conditions
+        obj_function = partial(mbo_wrapper._obj_function, base_estimator=classifier, cv_iter=list(StratifiedKFold(3).split(X, y)), X=X, y=y)
+        mbo_wrapper.fit(X, y)
+
+        # Smoke test the score etc:
+        mbo_wrapper.score(X, y)
+        mbo_wrapper.predict_proba(X)
+
+        # Test exception handling on scoring
+        mbo_wrapper.scoring = 'sklearn'
+        assert_raises(ValueError, mbo_wrapper.fit, X, y)
+
+        mbo_params = mbo_wrapper.best_estimator_.get_params()
+
+        # SMAC scenario oject
+        scenario = Scenario({"run_obj": "quality",  "runcount-limit": n_iter, "cs": config_space, "deterministic": "true", "memory_limit": 3072})
+
+        # To optimize, we pass the function to the SMAC-object
+        smac = SMAC(scenario=scenario, rng=random_seed,
+                    tae_runner=obj_function)
+        smac_incumbent = smac.optimize()
+        smac_inc_value = np.mean(np.array(smac.get_tae_runner().run(smac_incumbent, 1)[3]['test_scores']))
+
+        # note that multiple configurations could have led to the same
+        # SMAC always takes the most recently found # TODO: check assumption
```

Comment:
Basically, there are two TODO's. The other one is extending the function`_param_distributions_to_config_space` to support scipy ranges, which I consider future work. 

I removed them from the comments and raised appropriate errors.